src/components/NcSelectTags/api.js

```diff
@@ -21,95 +22,61 @@
  */
 
 import axios from '@nextcloud/axios'
-import { generateRemoteUrl } from '@nextcloud/router'
+import camelCase from 'camelcase'
+import { generateUrl } from '@nextcloud/router'
 
-const xmlToJson = (xml) => {
-	let obj = {}
+import { davClient } from './davClient.js'
+import logger from '../../utils/logger.js'
+import { t } from '../../l10n.js'
 
-	if (xml.nodeType === 1) {
-		if (xml.attributes.length > 0) {
-			obj['@attributes'] = {}
-			for (let j = 0; j < xml.attributes.length; j++) {
-				const attribute = xml.attributes.item(j)
-				obj['@attributes'][attribute.nodeName] = attribute.nodeValue
-			}
-		}
-	} else if (xml.nodeType === 3) {
-		obj = xml.nodeValue
-	}
+const fetchTagsBody = `<?xml version="1.0"?>
+<d:propfind  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
+	<d:prop>
+		<oc:id />
+		<oc:display-name />
+		<oc:user-visible />
+		<oc:user-assignable />
+		<oc:can-assign />
+	</d:prop>
+</d:propfind>`
 
-	if (xml.hasChildNodes()) {
-		for (let i = 0; i < xml.childNodes.length; i++) {
-			const item = xml.childNodes.item(i)
-			const nodeName = item.nodeName
-			if (typeof (obj[nodeName]) === 'undefined') {
-				obj[nodeName] = xmlToJson(item)
-			} else {
-				if (typeof obj[nodeName].push === 'undefined') {
-					const old = obj[nodeName]
-					obj[nodeName] = []
-					obj[nodeName].push(old)
-				}
-				obj[nodeName].push(xmlToJson(item))
-			}
-		}
-	}
-	return obj
+const parseTags = (tags) => {
+	return tags.map(({ props }) => Object.fromEntries(
+		Object.entries(props)
+			.map(([key, value]) => [camelCase(key), value])
+	))
 }
 
-const parseXml = (xml) => {
-	let dom = null
-	try {
-		dom = (new DOMParser()).parseFromString(xml, 'text/xml')
-	} catch (e) {
-		console.error('Failed to parse xml document', e)
+export const fetchTags = async () => {
```

Comment:
There are instances with thousands of tags, so loading all of them might not scale well and/or even break the ui

Reply:
This change does not affect the number of tags returned from the server endpoint which is the same as before

Any changes to the number of tags returned, i.e. pagination, would be out-of-scope

Is there something within the scope of this PR that you'd suggest?